### ISSUE TITLE
Listing folder globs gives SQL error

### ISSUE THREAD

**Issue by ewels** [ScilifelabDataCentre/dds_web#466]
Testing the `dds ls` command line tool I managed to get this:

```console
$ dds ls fac002 */_old
     ︵
 ︵ (  )   ︵
(  ) ) (  (  )   SciLifeLab Data Delivery System
 ︶  (  ) ) (    https://www.scilifelab.se/data
      ︶ (  )    Version 0.2.0
          ︶

INFO     Listing files for project 'fac002'                                                                                                                  data_lister.py:124
INFO     Showing files in folder '*/_old'                                                                                                                    data_lister.py:126
ERROR    Failed to get list of files: (pymysql.err.OperationalError) (1139, "Regex error 'quantifier does not follow a repeatable item at offset 1'")           __main__.py:265
         [SQL: SELECT DISTINCT files.subpath AS files_subpath
         FROM files
         WHERE files.project_id = binary(%(binary_1)s) AND (files.subpath regexp %(subpath_1)s)]
         +)+$'}]
         (Background on this error at: http://sqlalche.me/e/14/e3q8)
```

I haven't dug into the code to figure out what this really means yet, but getting a SQL error is slightly scary (I'm assuming that little Bobby Tables will be safe here..? 👀 ).

Anyway, would be nice to enable file path globbing when listing files if possible, or catching this type of error if not (and just returning a files-not-found response).

**Comment by inaod568** [comment]
Ah, we haven't added/tried the `*` stuff yet, but good that you noticed this.

**Comment by ewels** [comment]
I guess my main concern was not that the functionality was missing, but that I got a SQL command returned. I wanted to be sure that the back-end is not vulnerable to SQL injection attacks (hence my [Bobby Tables reference](https://xkcd.com/327/)).

I just had a quick poke and it's all SQLalchemy and no direct SQL command building as far as I can see though, so I guess it should be safe 👍🏻

**Comment by inaod568** [comment]
Yes, only SQLAlchemy 👍🏻 But it's a good point that we should look into a bit more

**Comment by inaod568** [comment]
Now `dds ls [PROJECT] */<anything here>` gives 

> Error: Got unexpected extra arguments

Closing this and creating a new card for adding the glob functionality.

**Comment by ewels** [comment]
That's a little surprising - was that with a glob path that existed locally? That would make sense as bash would expand the glob and pass multiple positional arguments.

I suspect that if you pass with a glob that doesn't match anything (or quote the glob) then the above error will still be there.

**Comment by inaod568** [comment]
Aah, ok. That makes sense. Apparently the path I tried did exist locally,
now that I tried another one I got the SQL error.

Den mån 14 juni 2021 kl 08:05 skrev Phil Ewels ***@***.***>:
